tools/debug_tool/templates/test.html

```diff
@@ -0,0 +1,250 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Page Title</title>
+<style>
+html, body {
+  height: 100%;
+  margin: 0;
+}
+
+.full-height {
+  height: 100%;
+  background: white;
+}
+</style>
+
+<!--link rel="stylesheet" href="format.css"-->
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+<!--script src="http://marvl.infotech.monash.edu/webcola/cola.js"></script-->
+<!--script src="https://ialab.it.monash.edu/webcola/cola.js"></script-->
+<script src="https://cdn.jsdelivr.net/gh/tgdwyer/WebCola/WebCola/cola.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/d3/3.4.1/d3.min.js"></script>
+</head>
+
+<!DOCTYPE Create chart on page>
+<body>
+<!-- <div id="chartContainer" style="height: 370px; width: 100%;"></div> -->
+<div id="graph" class="full-height"> Kotekan Pipeline</div>
+<script type="text/javascript" src="https://canvasjs.com/assets/script/jquery-1.11.1.min.js"></script>
+<script type="text/javascript" src="https://canvasjs.com/assets/script/jquery.canvasjs.min.js"></script>
+
+
+</body>
+
+<script>
+function updateChart(buffers) {
+
+    // Get buffer names
+    var bufNames = Object.keys(buffers);
+
+    // Populate datapoints of chart
+    var dps = [];
+    for (var val in buffers.network_buffer_0.frames) {
+      dps.push({y: buffers.network_buffer_0.frames[val]});     
+    }
+    
+    //Better to construct options first and then pass it as a parameter
+    var options = {
+    	title: {
+    		text: bufNames[11] + " Frames"              
+    	},
+    	data: [              
+    	{
+    		// Change type to "doughnut", "line", "splineArea", etc.
+    		type: "column",
+            dataPoints: dps
+    	}
+    	]
+    };
+    
+    $("#chartContainer").CanvasJSChart(options);
+}
+
+// Time between updating kotekan metrics
+const POLL_WAIT_TIME_MS = 2000;
+
+// Poll kotekan via web server every POLL_WAIT_TIME_MS and update page with new metrics
+async function poll() {
+  let response = await fetch("/update");
+  //console.log("Polling... " + response.json());
+
+  if (response.status == 502) {
+    // Status 502 is a connection timeout error,
+    // may happen when the connection was pending for too long,
+    // and the remote server or a proxy closed it
+    // let's reconnect
+  } else if (response.status != 200) {
+    // An error - let's show it
+    console.log("Error: " + response.statusText);
+    // Reconnect in one second
+    await new Promise(resolve => setTimeout(resolve, 1000));
+  } else {
+
+    // Get and show the message
+    let buffers = await response.json();
+    //console.log("Got response: " + message.network_buffer_0.frames);
+    //drawTree(buffers);
+  }
+
+  // Call poll() again to get the next message
+  setTimeout(() => { poll(); }, POLL_WAIT_TIME_MS);
+}
+
+// Start polling kotekan for metrics
+//poll();
```

Comment:
@Jintao-Xu this is the function I was talking about. It should poll the kotekan endpoint every POLL_WAIT_TIME_MS.

Reply:
And you would want to call your draw function where I call `drawTree(buffers)`.